server/src/test/java/org/elasticsearch/gateway/PersistedClusterStateServiceTests.java

```diff
@@ -1629,6 +1894,18 @@ private NodeEnvironment newNodeEnvironment(Path[] dataPaths) throws IOException
         );
     }
 
+    private static MappingMetadata randomMappingMetadata(boolean allowNull) {
```

Comment:
nit: slight preference for this being two methods `randomMappingMetadata` vs `randomMappingMetadataOrNull`